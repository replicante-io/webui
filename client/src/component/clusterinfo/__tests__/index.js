'use strict';
//@flow

import '@babel/polyfill';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { OverviewRedirect } from '../index';
import ClusterInfo from '../index';


const mockStore = configureStore([]);
const MATCH = {
  params: {name: 'test'},
  path: 'about:blank/clusters/test/',
  url: 'about:blank/clusters/:name/',
};


describe('ClusterInfo', () => {
  describe('index', () => {
    let store;

    beforeEach(() => {
      store = mockStore({
        clusterinfo: {
          discovery: {},
          meta: {}
        }
      });
    });

    test('renders index', () => {
      store = mockStore({ clusterinfo: { meta: {
        'test': {}
      } } });
      const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <ClusterInfo match={MATCH} />
          </Provider>
        </BrowserRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders index loading', () => {
      const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <ClusterInfo match={MATCH} />
          </Provider>
        </BrowserRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders redirect', () => {
      const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <OverviewRedirect match={MATCH} />
          </Provider>
        </BrowserRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
