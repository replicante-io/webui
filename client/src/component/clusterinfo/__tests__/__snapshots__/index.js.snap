// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterInfo index renders index 1`] = `
<div>
  <h1 />
  <div
    className="cluster-tabs"
  >
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/overview"
      onClick={[Function]}
    >
      Overview
    </a>
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/discovery"
      onClick={[Function]}
    >
      Discovery
    </a>
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/agents"
      onClick={[Function]}
    >
      Agents
    </a>
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/nodes"
      onClick={[Function]}
    >
      Nodes
    </a>
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/events"
      onClick={[Function]}
    >
      Events
    </a>
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/actions"
      onClick={[Function]}
    >
      Actions
    </a>
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/orchestrators-actions"
      onClick={[Function]}
    >
      Orchestrator Actions
    </a>
    <a
      aria-current={null}
      className="tab"
      href="/about:blank/clusters/:name/orchestrate-report"
      onClick={[Function]}
    >
      Orchestrate Report
    </a>
  </div>
  <div />
</div>
`;

exports[`ClusterInfo index renders index loading 1`] = `
<div
  className="text-center"
>
  <p
    className="large-icon nodata"
  >
    <span
      className="fa-layers fa-fw"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-database "
        data-icon="database"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 80v48c0 44.2-100.3 80-224 80S0 172.2 0 128V80C0 35.8 100.3 0 224 0S448 35.8 448 80zM393.2 214.7c20.8-7.4 39.9-16.9 54.8-28.6V288c0 44.2-100.3 80-224 80S0 332.2 0 288V186.1c14.9 11.8 34 21.2 54.8 28.6C99.7 230.7 159.5 240 224 240s124.3-9.3 169.2-25.3zM0 346.1c14.9 11.8 34 21.2 54.8 28.6C99.7 390.7 159.5 400 224 400s124.3-9.3 169.2-25.3c20.8-7.4 39.9-16.9 54.8-28.6V432c0 44.2-100.3 80-224 80S0 476.2 0 432V346.1z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-question overlay"
        data-icon="question"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c-17.7 0-32 14.3-32 32s-14.3 32-32 32s-32-14.3-32-32C0 75 43 32 96 32h97c70.1 0 127 56.9 127 127c0 52.4-32.2 99.4-81 118.4l-63 24.5 0 18.1c0 17.7-14.3 32-32 32s-32-14.3-32-32V301.9c0-26.4 16.2-50.1 40.8-59.6l63-24.5C240 208.3 256 185 256 159c0-34.8-28.2-63-63-63H96zm48 384c-22.1 0-40-17.9-40-40s17.9-40 40-40s40 17.9 40 40s-17.9 40-40 40z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </span>
  </p>
  No information available. 
  <br />
  If nothing shows up the cluster may not exist.
</div>
`;

exports[`ClusterInfo index renders redirect 1`] = `null`;
